packages/react-query/src/useBaseQuery.ts

```diff
@@ -78,57 +96,70 @@ export function useBaseQuery<
       ),
   )
 
+  // Duc - skip this in MVP
   const result = observer.getOptimisticResult(defaultedOptions)
 
+  // Duc - why do we need useSyncExternalStore
+  // need to check React document to find out
   useSyncExternalStore(
```

Comment:
this makes sure our observers are subscribed at the right time. it's react specific.